omod-common/src/main/java/org/openmrs/module/webservices/rest/web/ConversionUtil.java

```diff
@@ -539,56 +543,46 @@ public static DelegatingResourceDescription getCustomRepresentationDescription(C
 		String def = representation.getRepresentation();
 		def = def.startsWith("(") ? def.substring(1) : def;
 		def = def.endsWith(")") ? def.substring(0, def.length() - 1) : def;
```

Comment:
I don't know where this original code came from - I am shown on the blame in github, but I have no recollection writing this and it doesn't look like my style, so I probably moved it from somewhere and that history is lost.

In any event, this seems to work in some cases (obviously it must), but does not work for certain combinations of nested properties.  As can be seen below, the original parsing was pretty brittle, splitting on commas even though commas can be used to separate fields both at the top level and within nested objects, and the code is not resilient to all cases.